### ISSUE TITLE
How to run parallel tests using TestNG Listener with Extent Report?

### ISSUE THREAD

**Issue by ArmandsBruns** [anshooarora/extentreports-java#882]
I'm using Selenium Java and 2.41.2 of extent Report with TestNG Framework

Is there a good description link about it or examples?

I created two classes: ExtentManager and ExtentTestManager.

**Comment by anshooarora** [comment]
Please see http://extentreports.com/docs/versions/2/java/ for documentation and examples.  

Also,  I would recommend using version 3.0 as you are starting off as its what's being regularly maintained.  Version 3 docs:  http://extentreports.com/docs/versions/3/java/

**Comment by VasaviUppada** [comment]
Hi,
I need help on the same... When I'm running my scripts in a sequence I can able to run successfully & Im getting results too. When I'm running the same tests/methods/classes in parallel, all my Extent reports are getting messed up (one test logs are mixed into another test logs) and failing.

For my tests, Im using Selenium with TestNG, Maven & Im running my tests in Saucelabs.

Please help me on this

**Comment by harsh976** [comment]
@vasaviUppada 
Are yout you any listner of testng?
If yes then please share the code.
And also are you doing paralle tests or methods or classes

**Comment by VasaviUppada** [comment]
Hi Anush,

I tried in multiple ways to run in parallel (methods, tests, classes)
when ever I ran in parallel, all reports are messing up & if I remove
parallel, it is working fine.
Can you please provide me your personal mail id ... so that I can share my
code with you.
It will be more clearer for you to understand

**Comment by harsh976** [comment]
harsh976@gmail.com

**Comment by VasaviUppada** [comment]
Thanks for sharing email....
Can we create separate thread for extentTest to run each test? Will it resolve my current issue ?

**Comment by anshooarora** [comment]
Please open a new ticket.  Thank you.

**Comment by manishbhalshankar** [comment]
We are also running scripts sequentially as well as parallel and our report looks to be proper. Following the the way in which we generate the report:
1.  We have a TestNGListener setup which implements ITestListener, IReporter, ISuiteListener
2.  In Overridden method 'generateReport' we create an object of our Report class which initializes the ExtentReport and ExtentXReport and also has ExtentReports and ExtentTest fields.
3.  We than build the test nodes by passing the context passed, failed and skipped tests from suit-result-testcontext.

Hope I understood your issue properly and that these steps help you resolve the issue.
